lua/lspconfig/esbonio.lua

```diff
@@ -0,0 +1,58 @@
+local configs = require 'lspconfig/configs'
+local utils = require 'lspconfig/util'
+
+configs.esbonio = {
+  default_config = {
+    cmd = { 'python', '-m', 'esbonio' },
+    filetypes = { 'rst' },
+    root_dir = function(filename)
+      return utils.root_pattern('conf.py', '.git')(filename) or utils.path.dirname(filename)
+    end,
```

Comment:
`dirname` fallbacks is going away in favour of `single_file_support = true`, see:

- GH-1385

```suggestion
    root_dir = utils.root_pattern('conf.py', '.git'),
```